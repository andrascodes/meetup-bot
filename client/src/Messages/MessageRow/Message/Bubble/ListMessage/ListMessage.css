.ListMessage {
  background-color: #fff;
  color: #000;
  border-radius: 1em;
  border: 0.2em solid rgba(0, 0, 0, .10);
}

.ListElementContainer {
  border-bottom: 0.1em solid rgba(0, 0, 0, .10);
}

.ListElement {
  padding: 0.5em;
}

.ListElementTitle {
  color: #000;
  font-weight: bold;
}

.ListElementSubtitle {
  color: #999;
}

.ListElementImage {
  width: 100px;
  height: 100px;
  border: 0.1em solid rgba(0, 0, 0, .10);
  border-radius: 0.25em;
}

.ListElementButton {
  font-size: 13px;
  padding: 6px 8px 6px 8px;
  border: 0.1em solid #0084ff;
  border-radius: 0.25em;
  color: #0084ff;
  white-space: nowrap;
  display: inline-block;
  margin: 1em 0em;
}

.ListButton {
  padding: 0.3em 0em;
  color: #0084ff;
  font-weight: 500;
}