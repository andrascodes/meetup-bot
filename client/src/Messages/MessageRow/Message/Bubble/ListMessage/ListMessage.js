import React from 'react'

// import url from 'url';

import './ListMessage.css'

export const ListMessage = props => (
  <div className="ListMessage">
    {props.attachment.payload.elements.map((elem, ind) => (
      <div className="ListElementContainer" key={ind}>
        <div className="ListElement row">
          <div className="col-lg-6 col-md-6 col-sm-6 col-xs-6 text-left">
            <div className="ListElementTitle">{elem.title}</div>
            <div className="ListElementSubtitle">{elem.subtitle}</div>
            <div className="ListElementDefaultAction ListElementSubtitle">
              {elem.default_action && new URL(elem.default_action.url).hostname}
            </div>
            {elem.buttons && elem.buttons.map((btn, ind) => (
              <div className="ListElementButton" key={ind}>
                {btn.title}
              </div>
            ))}            
          </div>
          <div className="col-lg-6 col-md-6 col-sm-6 col-xs-6 text-right">
            <img className="ListElementImage" src={elem.image_url} alt=""/>
          </div>
        </div>
      </div>
    ))}

    {props.attachment.payload.buttons.map((btn, ind) => (
      <div key={ind} className="ListButton text-center">
        {btn.title}
      </div>
    ))}
  </div>
)