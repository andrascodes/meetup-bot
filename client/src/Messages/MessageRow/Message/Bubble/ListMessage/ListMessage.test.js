import React from 'react'
import renderer from 'react-test-renderer'

import { ListMessage } from './ListMessage'

describe.only('ListMessage', () => {

  // attachment.payload.elements: array
    // title, subtitle, default_action.url
    // buttons
      // title
    // image_url
  // buttons: title
  test('renders correctly', () => {
    const tree = renderer.create(
      <ListMessage
        attachment={{
          payload: {
            elements: [
              {
                title: 'Title 1',
                subtitle: 'Subtitle 1',
                default_action: {
                  url: 'https://www.google.com'
                },
                image_url: 'https://www.google.com',
                buttons: [
                  {
                    title: 'ElemBtn 1'
                  }
                ]
              },
              {
                title: 'Title 2',
                subtitle: 'Subtitle 2',
                default_action: {
                  url: 'https://www.google.com'
                },
                image_url: 'https://www.google.com',
                buttons: [
                  {
                    title: 'ElemBtn 2'
                  }
                ]
              },
            ],
            buttons: [
              {
                title: 'ListBtn'
              }
            ]
          }
        }}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

})