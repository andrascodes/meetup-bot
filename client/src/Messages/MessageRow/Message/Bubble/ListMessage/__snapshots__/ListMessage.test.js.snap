exports[`ListMessage renders correctly 1`] = `
<div
  className="ListMessage">
  <div
    className="ListElementContainer">
    <div
      className="ListElement row">
      <div
        className="col-lg-6 col-md-6 col-sm-6 col-xs-6 text-left">
        <div
          className="ListElementTitle">
          Title 1
        </div>
        <div
          className="ListElementSubtitle">
          Subtitle 1
        </div>
        <div
          className="ListElementDefaultAction ListElementSubtitle">
          www.google.com
        </div>
        <div
          className="ListElementButton">
          ElemBtn 1
        </div>
      </div>
      <div
        className="col-lg-6 col-md-6 col-sm-6 col-xs-6 text-right">
        <img
          alt=""
          className="ListElementImage"
          src="https://www.google.com" />
      </div>
    </div>
  </div>
  <div
    className="ListElementContainer">
    <div
      className="ListElement row">
      <div
        className="col-lg-6 col-md-6 col-sm-6 col-xs-6 text-left">
        <div
          className="ListElementTitle">
          Title 2
        </div>
        <div
          className="ListElementSubtitle">
          Subtitle 2
        </div>
        <div
          className="ListElementDefaultAction ListElementSubtitle">
          www.google.com
        </div>
        <div
          className="ListElementButton">
          ElemBtn 2
        </div>
      </div>
      <div
        className="col-lg-6 col-md-6 col-sm-6 col-xs-6 text-right">
        <img
          alt=""
          className="ListElementImage"
          src="https://www.google.com" />
      </div>
    </div>
  </div>
  <div
    className="ListButton text-center">
    ListBtn
  </div>
</div>
`;
